--- a/xbmc/interfaces/legacy/ModuleXbmcgui.cpp	2014-06-05 14:48:24.000000000 +0300	
+++ b/xbmc/interfaces/legacy/ModuleXbmcgui.cpp	2014-06-06 14:48:24.000000000 +0300
@@ -25,2 +25,3 @@
 #include "guilib/GUIWindowManager.h"
+#include "input/MouseStat.h"
 #include "utils/log.h"
@@ -54,2 +55,23 @@
     }
+
+    long getMouseRawPosition()
+    {
+      return g_Mouse.GetRawX() * 10000 + g_Mouse.GetRawY();
+    }
+     
+    long getMousePosition()
+    {
+      return g_Mouse.GetX() * 10000 + g_Mouse.GetY();
+    }
+     
+    long getMouseRawAction()
+    {
+    	return g_Mouse.GetRawAction();
+    }
+
+    long getMouseAction()
+    {
+    	return g_Mouse.GetAction();
+    }
+
   }
